salt/utils/schedule.py

```diff
@@ -580,6 +598,18 @@ def eval(self):
             now = int(time.time())
             time_conflict = False
 
+            if 'until' in data:
+                if not _WHEN_SUPPORTED:
+                    log.error('Missing python-dateutil.'
+                              'Ignoring until.')
+                else:
+                    until__ = dateutil_parser.parse(data['until'])
+                    until = int(time.mktime(until__.timetuple()))
+
+                    if until <= now:
+                        log.debug('Until time has passed skipping job.')
```

Comment:
Could we maybe add some info here on which job is being skipped? That might be helpful. :]